### PR TITLE
[ML] Azure standards related fixes for Job Operations

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py [hunk 1]
```diff
@@ -921,6 +942,7 @@ def _resolve_job_input(self, entry: Union[Input, str, bool, int, float], base_pa
                     self._datastore_operations,
                     entry.path,
                     datastore_name=datastore_name,
+                    show_progress=self._show_progress,
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:945]: Is this change intentional ?

Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py]: reverted hanchi's change

Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py]: This is reverted

**Review by DouglasXiaoMS — APPROVED** [review]

**Review by needuv — APPROVED** [review]
